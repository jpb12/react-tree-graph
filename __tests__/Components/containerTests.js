import React from 'react';
import { shallow } from 'enzyme';

import Container from '../../src/components/container';
import Link from '../../src/components/link';
import Node from '../../src/components/node';

const nodes = [
	{
		x: 1,
		y: 2,
		data: {
			name: 'Colour'
		}
	}, {
		x: 100,
		y: 50,
		data: {
			name: 'Black'
		}
	}
];

const defaultProps = {
	nodes: [],
	links: [],
	height: 100,
	keyProp: 'name',
	labelProp: 'name',
	nodeOffset: 3.5,
	nodeRadius: 5,
	width: 200,
	circleProps: {},
	gProps: {},
	pathProps: {},
	svgProps: {},
	textProps: {}
};

describe('<Container>', () => {
	test('renders correctly', () => {
		const props = {
			nodes: nodes,
			links: [{
				source: nodes[0],
				target: nodes[1]
			}]
		};

		const tree = shallow(<Container {...defaultProps} {...props}/>);

		expect(tree).toMatchSnapshot();
	});

	test('renders children', () => {
		const props = {
			nodes: nodes,
			links: [{
				source: nodes[0],
				target: nodes[1]
			}]
		};

		const tree = shallow(<Container {...defaultProps} {...props}><text>Extra child</text></Container>);

		expect(tree).toMatchSnapshot();
	});

	test('html tree props added', () => {
		const props = {
			svgProps: {
				className: 'test-class',
				stoke: 'none'
			}
		};

		const tree = shallow(<Container {...defaultProps} {...props}/>);

		expect(tree).toMatchSnapshot();
	});

	test('path props combined', () => {
		const props = {
			links: [{

				source: nodes[0],
				target: { ...nodes[1], data: { name: 1 } }
			}, {
				name: 2,
				source: nodes[0],
				target: { ...nodes[1], data: { name: 1, pathProps: { className: 'override' } } }
			}],
			pathProps: {
				className: 'default'
			}
		};

		const tree = shallow(<Container {...defaultProps} {...props}/>);

		const links = tree.find(Link);
		expect(links.length).toBe(2);
		expect(links.at(0).props().pathProps).toEqual({ className: 'default' });
		expect(links.at(1).props().pathProps).toEqual({ className: 'override' });
	});

	test('node props combined', () => {
		const props = {
			nodes: [
				{ ...nodes[0], data: { name: 1, gProps: { className: 'default' } } },
				{ ...nodes[1], data: { name: 2, gProps: { className: 'override' } } }
			]
		};

		const tree = shallow(<Container {...defaultProps} {...props}/>);

		const domNodes = tree.find(Node);
		expect(domNodes.length).toBe(2);
		expect(domNodes.at(0).props().gProps).toEqual({ className: 'default' });
		expect(domNodes.at(1).props().gProps).toEqual({ className: 'override' });
	});
});