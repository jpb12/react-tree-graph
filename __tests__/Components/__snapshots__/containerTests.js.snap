// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<Container> html tree props added 1`] = `
<svg
  className="test-class"
  height={100}
  stoke="none"
  width={200}
>
  <g
    transform="translate(20, 10)"
  />
</svg>
`;

exports[`<Container> renders children 1`] = `
<svg
  height={100}
  width={200}
>
  <text>
    Extra child
  </text>
  <g
    transform="translate(20, 10)"
  >
    <Link
      key="Black"
      keyProp="name"
      pathProps={{}}
      source={
        {
          "data": {
            "name": "Colour",
          },
          "x": 1,
          "y": 2,
        }
      }
      target={
        {
          "data": {
            "name": "Black",
          },
          "x": 100,
          "y": 50,
        }
      }
      x1={1}
      x2={100}
      y1={2}
      y2={50}
    />
    <Node
      direction="ltr"
      gProps={{}}
      key="Colour"
      keyProp="name"
      labelProp="name"
      name="Colour"
      nodeProps={{}}
      shape="circle"
      textProps={{}}
      x={1}
      y={2}
    />
    <Node
      direction="ltr"
      gProps={{}}
      key="Black"
      keyProp="name"
      labelProp="name"
      name="Black"
      nodeProps={{}}
      shape="circle"
      textProps={{}}
      x={100}
      y={50}
    />
  </g>
</svg>
`;

exports[`<Container> renders correctly 1`] = `
<svg
  height={100}
  width={200}
>
  <g
    transform="translate(20, 10)"
  >
    <Link
      key="Black"
      keyProp="name"
      pathProps={{}}
      source={
        {
          "data": {
            "name": "Colour",
          },
          "x": 1,
          "y": 2,
        }
      }
      target={
        {
          "data": {
            "name": "Black",
          },
          "x": 100,
          "y": 50,
        }
      }
      x1={1}
      x2={100}
      y1={2}
      y2={50}
    />
    <Node
      direction="ltr"
      gProps={{}}
      key="Colour"
      keyProp="name"
      labelProp="name"
      name="Colour"
      nodeProps={{}}
      shape="circle"
      textProps={{}}
      x={1}
      y={2}
    />
    <Node
      direction="ltr"
      gProps={{}}
      key="Black"
      keyProp="name"
      labelProp="name"
      name="Black"
      nodeProps={{}}
      shape="circle"
      textProps={{}}
      x={100}
      y={50}
    />
  </g>
</svg>
`;
